src/main/java/org/klodnicki/service/AccountService.java

```diff
@@ -47,13 +47,114 @@ private List<Account> getAllAccountsFromDatabase() {
+            throw new NotFoundInDatabaseException(Account.class);
+        }
+
+        if (parameterToModify.equalsIgnoreCase(SortOptionAccount.PESEL.getSortName()) && newData.length() != 11) {
+            throw new IllegalArgumentException("Pesel must have 11 digits.");
+
+        }
+
+        accountRepository.mergeAccount(peselAccountToModify, prepareParameterToDatabase(parameterToModify), newData);
+    }
+
+    private String prepareParameterToDatabase(String parameter) {
+
+        for (SortOptionAccount sortOption : SortOptionAccount.values()) {
+            if (sortOption.getSortName().equalsIgnoreCase(parameter)) {
+                parameter = sortOption.getHqlParameter();
+            }
+        }
+        return parameter;
+    }
+
+    public List<String> sortOptionNames() {
+        List<String> namesList = new ArrayList<>();
+        for (SortOptionAccount sortOption : SortOptionAccount.values()) {
+            namesList.add(sortOption.getSortName());
+        }
+        return namesList;
+    }
+
+    public List<String> prepareListOfAccountsByFirstNameAndLastName(String firstName, String lastName)
+            throws NotFoundInDatabaseException {
+        List<Account> accountsByFirstNameAndLastName = accountRepository.findAccountsByFirstNameAndLastName
+                (firstName, lastName);
+        List<String> results = new ArrayList<>();
+
+        if (accountsByFirstNameAndLastName.isEmpty()) {
+            throw new NotFoundInDatabaseException(Account.class);
+        }
+
+        for (Account account : accountsByFirstNameAndLastName) {
+            results.add(account.toString());
+        }
+        return results;
+    }
+
+    public List<String> prepareAllAccountsOrderByParameter(String parameter) {
+        List<Account> accountsOrderByParameter = accountRepository.findAllAccountsOrderByParameter
+                (prepareParameterToDatabase(parameter));
+        List<String> results = new ArrayList<>();
+
+        for (int i = 0; i < accountsOrderByParameter.size(); i++) {
+            results.add(accountsOrderByParameter.get(i).toString());
+        }
+        return results;
+    }
+
+    private boolean peselExist(String pesel, String firstName, String lastName) {
+        List<Account> accountList = accountRepository.findAccountsByFirstNameAndLastName(firstName, lastName);
+
+        for (int i = 0; i < accountList.size(); i++) {
+            if (pesel.equalsIgnoreCase(accountList.get(i).getPesel())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public List<String> prepareAccountByPeselWithBooks(String firstName, String lastName, String peselAccount)
+            throws NotFoundInDatabaseException {
+        List<String> results = new ArrayList<>();
+        List<Account> preparedList = accountRepository.findAccountsByFirstNameAndLastName(firstName, lastName);
+
+        for (int i = 0; i < preparedList.size(); i++) {
```

Comment:
for me like is now it is more readable, but if you insist I can change that into for each loop

Reply:
I wrote this comment because it is now not that readable (could be better). You're using `preparedList.get(i).getBooks().get(j)` and `preparedList.get(i)` three times in a row each which just looks bad.
Foreach is one alternative that will make it look cleaner and more readable for people checking the code, trust me :)

Please, do not resolve conversations like that! I mean when I'm asking you to change something and you just say that looks ok to you. Resolved conversation are when we both came in the end to some conclusion, not when you just respond to my comment.

Reply:
OK! done!